packages/pdf-viewer/hooks/useIsVisible.ts

```diff
@@ -1,8 +1,8 @@
-import React, { useEffect, useState } from 'react';
+import { useEffect, useState, MutableRefObject } from 'react';
 
 
-const useIsVisible = (elementRef: React.MutableRefObject<HTMLElement>, rootRef: React.MutableRefObject<HTMLElement>) => {
-	const [isVisible, setIsVisible] = useState(false);
+const useIsVisible = (elementRef: MutableRefObject<HTMLElement>, rootRef: MutableRefObject<HTMLElement>) => {
+	const [isVisible, setIsVisible] = useState(null);
```

Comment:
Wouldn't that be a boolean? Why "null"?

Reply:
`null` at first because some pages are always visible at the beginning (for ex: page 1), it would be wrong to call them `false` without checking first, `null` signifies we don't know yet. Later on when thumbnail will be added, it would cause an issue. But for now it would work fine either ways.

Reply:
Ok but you didn't update the calling code, so if you return null the calling code handles that as "false" anyway. I don't really get what you're doing.

And if it's null, true or false, it's no longer a boolean so if you really have to have that "null" state, please call this "visibleState" and make it an enum - Unknown, Invisible, Visible

Reply:
I'm keeping it at false for now, having 3 different state is causing performance issues if the pdf has over 2k pages. 
I will try to figure out something better when I create the thumbnails PR.